src/main/java/book/shop/service/impl/BookServiceImpl.java

```diff
@@ -1,24 +1,42 @@
 package book.shop.service.impl;
 
+import book.shop.dto.BookDto;
+import book.shop.dto.CreateBookRequestDto;
+import book.shop.mapper.BookMapper;
 import book.shop.model.Book;
 import book.shop.repository.BookRepository;
 import book.shop.service.BookService;
+import exception.EntityNotFoundException;
 import java.util.List;
+import java.util.Random;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn("XXX" + new Random().nextInt());
```

Comment:
it's better to remove for now, in future tasks you can add validation pattern or generaye isbn method